packages/components/src/entities/SamplesDeriveButton.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { FC, memo } from 'react';
+import { PermissionTypes } from '@labkey/api';
+
+import { QueryModel } from '../public/QueryModel/QueryModel';
+import { RequiresPermission } from '../internal/components/base/Permissions';
+import { ResponsiveMenuButton } from '../internal/components/buttons/ResponsiveMenuButton';
+import { SampleCreationType } from '../internal/components/samples/models';
+import { MAX_EDITABLE_GRID_ROWS } from '../internal/constants';
+import { DisableableButton } from '../internal/components/buttons/DisableableButton';
+
+import { CreateSamplesSubMenu, CreateSamplesSubMenuProps } from './CreateSamplesSubMenu';
+
+export interface SamplesDeriveButtonProps
+    extends Omit<
+        CreateSamplesSubMenuProps,
+        'id' | 'menuText' | 'parentQueryModel' | 'selectedQueryInfo' | 'selectedType' | 'subMenuText'
+    > {
+    asSubMenu?: boolean;
+    model: QueryModel;
+}
```

Comment:
I get what this is doing and the end result is the same so it makes sense. But I'm just curious of the benefit of this over the previous implementation. Previously this component's props interface defined the 6 props it needed. Now this is defining 2 and extending something that it then needs to omit 6 props from. What happens when a new prop is added to `CreateSamplesSubMenuProps` that isn't needed here?

Or is the reason for this type of extends because of the passthrough of those props to the child components below in the render/return? So is the rule of thumb that if you are passing through props via `{...props}` then the props of the current component should use extends/Omit?

Reply:
The latter but not exactly. I was following what the implementation was giving here as it was already supplying all of it's `props` to the `CreateSamplesSubMenu` components it instantiates. We certainly don't need to extend/Omit here and just support a subset of the properties but we shouldn't be declaring those props (of another component) again if we can avoid it. The other way to go is to extend/Pick where we pick the properties we want from the underlying object. Lastly, When passing through props we should always clean out props that are not part of the underlying components (hence `{ model, asSubMenu, ...createSampleMenuProps }`.

Reply:
Yup, that makes sense. Thanks for clarifying.